crates/rune/src/indexing/index.rs

```diff
@@ -285,6 +289,46 @@ impl<'a> Indexer<'a> {
         }))
     }
 
+    /// Expand a macro returning the current file
+    fn expand_file_macro(&mut self, ast: &mut ast::MacroCall) -> Result<BuiltInMacro, ParseError> {
+        let id = self.storage.insert_str(self.source.name());
```

Comment:
Rune sources can be completely in-memory, and names do not strictly speaking need to match the path. The best approach here is probably to use `self.source.path()` and error in case the source doesn't have a path.

Reply:
Also, principle this could use the `ast::Lit::new_with` constructor. But I only think it works for strings and a few more for now.

Reply:
I'm not sure if that'd make sense as a user, though. I'd expect it to expand to *something* in all builds given that we organize code based on a filesystem hierarchy... I'd maybe argue that if anything, the fact that the location is a filesystem path is unnecessary after loading. It's just code from a named source whether that be a webpage, a ZIP archive or a file on disk.

Reply:
All right. I'm fine either way really. Just rubs me the wrong way that `file!` does not mean a file semantically. But it's not a big deal since I expect it to only be used diagnostically!